web/src/lib/components/pages/record.tsx

```diff
@@ -158,11 +199,21 @@ export default class RecordPage extends Component<RecordProps, RecordState> {
   }
 
   private updateVolume(volume: number) {
-    if (!this.state.recording || !this.props.onVolume) {
+    if (!this.state.recording) {
       return;
     }
 
-    this.props.onVolume(volume);
+    // For some reason, volume is always exactly 100 at the end of the
+    // recording, even if it is silent; so ignore that.
+    if (volume !== 100 && volume > this.state.maxVolume) {
+      this.setState({
```

Comment:
Let's make `maxVolume` a property of the Record component, and not RecordState. RecordState is only for things that effect how the component is displayed, and since we only use `maxVolume` for validation checks, it doesn't belong in the state. This also has the side effect of triggering a render when we don't need to.